Hex Paper/DocumentsViewController.swift

```diff
@@ -0,0 +1,165 @@
+    private var initialRightBarButtonItems: [UIBarButtonItem]?
+    private var initialLeftBarButtonItems: [UIBarButtonItem]?
+
+    @IBOutlet weak var addButton: UIBarButtonItem!
+    
+    @IBAction func addButtonAction(_ sender: UIBarButtonItem) {
+        delegate?.documentsSceneAddButtonWasTapped(self)
+    }
+    
+    @IBAction func trashButtonAction(_ sender: UIBarButtonItem) {
+        trashButton.isEnabled = false
+        delegate?.documentsSceneTrashButtonWasTapped(self)
+    }
+    
+    @IBAction func doneButtonAction(_ sender: UIBarButtonItem) {
+        editMode = false
+    }
+
+    @IBAction func editButtonAction(_ sender: UIBarButtonItem) {
+        editMode = true
+    }
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        delegate = DocumentsController()
+        collectionView?.allowsSelection = true
+        initialRightBarButtonItems = navigationItem.rightBarButtonItems
+        initialLeftBarButtonItems = navigationItem.leftBarButtonItems
+        
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done,
+                                         target: self,
+                                         action: #selector(DocumentsViewController.doneButtonAction))
+        self.doneButton = doneButton
+        let trashButton = UIBarButtonItem(barButtonSystemItem: .trash,
+                                          target: self,
+                                          action: #selector(DocumentsViewController.trashButtonAction))
+        self.trashButton = trashButton
+    }
+
+}
+
+// Scene methods
+extension DocumentsViewController: DocumentsScene {
+    
+    var indexPathsForSelectedItems: [IndexPath]? {
+        get {
+            return collectionView?.indexPathsForSelectedItems
+        }
+    }
+    
+    func segueToEditDocumentScene() {
+        performSegue(withIdentifier: "editDocument", sender: self)
+    }
+    
+    func refreshDocumentData() {
+        collectionView?.reloadData()
+    }
+    
+    func deleteItems(at indexPaths: [IndexPath]) {
+        collectionView?.deleteItems(at: indexPaths)
+    }
+    
+}
+
+// UICollectionViewDelegate methods
+extension DocumentsViewController {
+    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if editMode {
+            trashButton.isEnabled = true
+            let cell = collectionView.cellForItem(at: indexPath)
+            cell?.layer.borderWidth = selectedBorderWidth
+        } else {
+            segueToEditDocumentScene()
+        }
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        if editMode {
+            let cell = collectionView.cellForItem(at: indexPath)
+            cell?.layer.borderWidth = borderWidth
+            if collectionView.indexPathsForSelectedItems == nil ||
+                collectionView.indexPathsForSelectedItems?.count == 0 {
```

Comment:
The check for nil seems redundant because `collecitonView.indexPathsForSelectedItems?` would take care of that.